vignettes/civis_ml.Rmd

```diff
@@ -0,0 +1,287 @@
+---
+title: "Modeling with CivisML"
+author: "Patrick Miller"
+date: "2017-08-29"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{civis_ml}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+CivisML is a machine learning service on Civis Platform. It allows you to fit many different models, do extensive meta-parameter tuning, and score data sets with millions of rows on AWS rather than your local machine. Results and plots are stored in Civis Platform, and can be easily shared with decision makers. The models themselves can also be stored, re-run, scheduled, and shared with other data scientists. [This blog post](https://www.civisanalytics.com/blog/civisml-scikit-learn-at-scale/) provides an excellent summary of the rich set of features in CivisML.
+
+CivisML is built in Python using scikit-learn. However, most of its features can be used through R without installing Python or any other dependencies using the `civis_ml` function in `civis`. It's possible to write sophisticated modeling workflows by combining `civis_ml` with other IO, reporting, and API functions in `civis`. 
+
+In the following, we will highlight the modeling feautures of `civis_ml`. While `civis_ml` is a complex function with many arguments, basic machine learning modeling and scoring can be easily carried out easily. We illustrate several features of `civis_ml` with data from a fictious company called Brandable, who is looking to predict which customers are likely to upgrade from the free to the premium service.
+
+## Data sources
+
+The first step of modeling with `civis_ml` is to specify the data source, which is the first argument. `civis_ml` works with local data frames, a CSV on local disk, tables in Redshift, and files on S3 (the files endpoint):
+
+```{r, eval = FALSE}
+library(civis)
+civis_ml(df, ...)
+civis_ml("path/to/data.csv", ...)
+civis_ml(civis_table(table_name = "schema.table", database_name = "database"), ...)
+civis_ml(civis_file(1234), ...)
+```
+
+The Brandable data is located in a Redshift table called `sample_project.premium_training_set`:
+
+```{r, eval  = FALSE}
+tab <- civis_table("sample_project.premium_training_set", "redshift-general")
```

Comment:
maybe something more clear like `my_database`?  I also like the approach [here](https://github.com/civisanalytics/civis-python/blob/master/examples/CivisML.ipynb) of dynamically getting the database name.